embed-code-go/embed_code.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2024, TeamDev. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+package main
+
+import (
+	"embed-code/embed-code-go/cli"
+	"embed-code/embed-code-go/configuration"
+	"flag"
+	"fmt"
+
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Struct with user-specified flags.
+//
+// codeRoot — a path to a root directory with code files.
+//
+// docsRoot — a path to a root directory with docs files.
+//
+// configPath — a path to a yaml configuration file which contains the roots.
+//
+// checkUpToDate — true to check for code embeddings to be up-to-date. Otherwise, the embedding is performed.
+type flags struct {
+	codeRoot      string
+	docsRoot      string
+	configPath    string
+	checkUpToDate bool
+}
+
+// Struct with roots that contained in a yaml configuration file.
+//
+// codeRoot — a path to a root directory with code files.
+//
+// docsRoot — a path to a root directory with docs files.
+type configFields struct {
```

Comment:
These are not `configFields`. These are paths.

Because of this name, the following code reads weird:
```
fun readRootsFromConfig(configPath string) configFields
```